### ISSUE TITLE
Optimize image to reduce size

### ISSUE THREAD

**Issue by Eldenroot** [mybbgroup/Image-auto-resizer#3]

**Comment by Eldenroot** [comment]
As I see, I am not the only one who share this idea :)

Are there any libraries to do that in php? @lairdshaw

**Comment by lairdshaw** [comment]
I haven't looked into it enough to know how effective they are at improving compression or otherwise optimising images, but there _are_ some libraries built in to PHP which manipulate images. There are also some third-party libraries coded in PHP which mostly seem to rely on other commandline tools or libraries, but those commandline toools or libraries wouldn't be installed on most web hosts.

**Comment by Eldenroot** [comment]
Yes, we should keep and try to keep wide compatibility for most common webhostings... so not to use any exotic modules which are not used in 99 % of cases.

Some inbuilt solution would be great...

**Comment by effone** [comment]
I just headed here to create an issue about compression and here is my mind reader.
I have a ready function made (using GD only) that can be implemented for compression. I will do some R&D and probably come up with a PR tonight.

**Comment by lairdshaw** [comment]
Sounds promising!

**Comment by Eldenroot** [comment]
@effone - sure, go ahead :) thx!
